File: src/main/java/exnihilocreatio/registries/types/FluidItemFluid.java
package exnihilocreatio.registries.types;

import exnihilocreatio.util.StackInfo;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;

@AllArgsConstructor
@EqualsAndHashCode
public class FluidItemFluid {

    @Getter
    private String inputFluid;

    @Getter
    private StackInfo reactant;

    @Getter
    private String output;
}

File: src/main/java/exnihilocreatio/registries/types/FluidItemFluid.kt
package exnihilocreatio.registries.types

import exnihilocreatio.util.StackInfo

// TODO: MOVE TO API PACKAGE ON BREAKING VERSION CHANGE
data class FluidItemFluid(
        val inputFluid: String? = null,
        val reactant: StackInfo? = null,
        val output: String? = null
)
